docs/Guides-Renderer.md

```diff
@@ -0,0 +1,168 @@
+
+## render prop
+
+We can change this behavior by supplying the `render` prop:
+
+```
+<Relay.Renderer
+  Container={ProfilePicture}
+  queryConfig={profileRoute}
+  render={({props}) => {
+      if (props) {
+        return <ProfilePicture {...props} />;
+      } else {
+        return <div>Loading...</div>;
+      }
+  }}
+/>
+```
+
+This snippet configures **Relay.Renderer** to render the "Loading..." text whenever it needs to fetch data.
+
+The `render` callback is called with an object with the following properties:
+
+```
+render({
+  props: ?{[propName: string]: mixed},
+  done: boolean,
+  error: ?Error,
+  retry: ?Function,
+  stale: boolean
+}): ?React$Element
+```
+
+### props
+
+```
+props: ?{[propName: string]: mixed}
+```
+
+If present, sufficient data is ready to render the container. This object must be spread into the container using the spread attribute operator. If absent, there is insufficient data to render the container.
+
+### done
+
+```
+done: boolean
+```
+
+Whether all data dependencies have been fulfilled. If `props` is present but `done` is false, then sufficient data is ready to render, but some data dependencies have not yet been fulfilled.
+
+### error
+
+```
+error: ?Error
+```
+
+If present, an error occurred while fulfilling data dependencies. If `props` and `error` are both present, then sufficient data is ready to render, but an error occurred while fulfilling deferred dependencies.
+
+### retry
+
+```
+retry: ?Function
+```
+
+A function that can be called to re-attempt to fulfill data dependencies. This property is only present if an `error` has occurred.
+
+### stale
+
+```
+stale: boolean
+```
+
+True when `forceFetch` is enabled and `props` is present due to available client data. Once the forced server request completes, `stale` will return to false.
+
+If a `render` callback is not supplied, the default behavior is to render the container if data is available, the existing view if one exists, or nothing.
+
+A `render` callback can simulate the default behavior by returning `undefined` to continue rendering the last view rendered (e.g. when transitioning from one `queryConfig` to another). Notice that this is different from a `render` callback that returns `null`, which would render nothing whenever data is loading, even if there was a previous view rendered.
+
+### Example
+
+```{4-6}
+// In this example, `ErrorComponent` and `LoadingComponent`
+// simply display a static error message / loading indicator.
+<Relay.Renderer
+  Container={ProfilePicture}
+  queryConfig={profileRoute}
+  render={({done, error, props, retry, stale}) => {
+        if (error) {
+          return <ErrorComponent />;
+        } else if (props) {
+          return <ProfilePicture {...props} />;
+        } else {
+          return <LoadingComponent />;
+        }
+      }}
+/>
+```
+
+### ref
```

Comment:
I think this needs two hashes, since your back outside of the “Render callback” section:

```
## Refs
```